samples/scripts/vnet-gateway.fsx

```diff
@@ -0,0 +1,45 @@
+
+#r @"./libs/Newtonsoft.Json.dll"
+#r @"../../src/Farmer/bin/Debug/netstandard2.0/Farmer.dll"
+
+open Farmer
+open Farmer.Builders
+open Farmer.VirtualNetworkGateway
+
+let privateNet = vnet {
+    name "my-vnet"
+    add_address_spaces [
+        "10.30.0.0/16"
+    ]
+    add_subnets [
+        subnet {
+            name "GatewaySubnet"
+            prefix "10.30.254.0/28"
+        }
+    ]
+}
+
+let gatewayIp = {
+    Name = ResourceName "gw-pip"
+    Location = Location.NorthEurope
+    DomainNameLabel = None
+}
+
+let myGateway = gateway {
+    name "er-gateway"
+    er_gateway_sku ErGatewaySku.Standard
+    vpn_type VpnType.RouteBased
+    vnet "my-vnet"
+    gateway_ip_config DynamicPrivateIp gatewayIp.Name.Value
```

Comment:
Is it worth collapsing the `gatewayIP` into the gateway itself? I'm wondering (and this is lack of knowledge showing here) what configuration is actually being set here - there's no domain label so what is that actually doing?

If it genuinely does need to be a separate type from the user's point of view, consider adding an overload for `gateway_ip_config` to take in a `GatewayIp` rather than needing to specify `Name.Value` (or at the very least, take in the `ResourceName` rather than the raw string and if needed, unwrap it in the builder).